src/views/Graph.vue

```diff
@@ -689,26 +745,39 @@ export default {
      * @param {Object[]} edges
      * @param {{ [id: string]: SVGRect }} nodeDimensions
      */
-    async layout (nodes, edges, nodeDimensions) {
+    async layout (nodes, edges, nodeDimensions, cycleSubgraph) {
       // generate the GraphViz dot code
-      const dotCode = this.getDotCode(nodeDimensions, nodes, edges)
+      const dotCode = this.getDotCode(nodeDimensions, nodes, edges, cycleSubgraph)
 
       // run the layout algorithm
       const jsonString = (await this.graphviz).layout(dotCode, 'json')
       const json = JSON.parse(jsonString)
 
+      this.subgraphs = {}
       // update graph node positions
       for (const obj of json.objects) {
-        const [x, y] = obj.pos.split(',')
-        const bbox = nodeDimensions[obj.name]
-        // translations:
-        // 1. The graphviz node coordinates
-        // 2. Centers the node on this coordinate
-        // TODO convert (2) to maths OR fix it to avoid recomputation?
-        this.nodeTransformations[obj.name] = `
+        if (obj.bb) {
+        // // if the object is a subgraph
+          this.subgraphs[obj.name] = {
+            x: obj.bb.split(',')[0],
+            y: -obj.bb.split(',')[3],
+            width: obj.bb.split(',')[2] - obj.bb.split(',')[0],
+            height: obj.bb.split(',')[3] - obj.bb.split(',')[1],
+            label: obj.label
```

Comment:
Is this correct? If so, it's more readable IMO. Not sure why `y = -top` though
```suggestion
          // if the object is a subgraph
          const [left, bottom, right, top] = obj.bb.split(',')
          this.subgraphs[obj.name] = {
            x: left,
            y: -top,
            width: right - left,
            height: top - bottom,
            label: obj.label
```

Reply:
This was a lot of trial and error tbh but without the - 
![image](https://github.com/cylc/cylc-ui/assets/7523921/19a3c616-cdca-4212-b05d-59252a3637b9)

Reply:
Example of what comes back from graphviz:
` { "x": "7604", "y": -734.8, "width": 4025, "height": 676.8, "label": "20000104T1200Z" } `

Reply:
Some systems disagree on the y axis, presumably GraphViz and SVG are an example. I think we do `-y` somewhere else in the graph view too.